### ISSUE TITLE
[Feat]: Manage Oauth with /preview links like /view links

### ISSUE THREAD

**Issue by lanedsmu** [lowcoder-org/lowcoder#1386]
When browsing to lowcoderurl/apps/appID/view, authentication happens as expected:  user is redirected to the appropriate auth provider (oauth or email or whatever) and then back to the app url after authentication.

With /preview URLs (and with URLs that don't have anything after the app ID), however, the behavior appears not to have been updated, and the user is redirected in ways I don't quite understand yet.  Behavior definitely does not mirror the behavior in /view.

Instead, we'd like the login behavior to be consistent, and to match what's done with /view URLs.

Additionally, it may be worth pointing out that when going to a particular app URL, there may not actually be a reason to prompt for an email address before taking the user to the auth screen: with an app specified in the URL, there's a defined destination org, so the email prompt is a bit superfluous.

**Comment by FalkWolsky** [comment]
You are right! These are truly good findings and we will work it out soon!
